### ISSUE TITLE
Batch Changes: 504 Gateway time-out

### ISSUE THREAD

**Issue by Jamesking56** [sourcegraph/sourcegraph#28252]
- **Sourcegraph version:** sourcegraph.com
- **Platform information:** MacOS 12.0.1 Monterey, zsh using src-cli version 3.34.1

#### Steps to reproduce:

I am following the Quickstart for Batch Changes guide here: https://docs.sourcegraph.com/batch_changes/quickstart

1. Setup src-cli with an account access token as per guide
2. Create batch spec called `phpstan-baseline-composer.batch.yaml` as follows:
```
name: phpstan-baseline-composer
description: Adds PHPStan baseline scripts to composer.json

on:
  - repositoriesMatchingQuery: file:composer.json

steps:
  - run: |
      jq '.scripts + {"phpstan-baseline": "vendor/bin/phpstan analyse --ansi --generate-baseline=phpstan-baseline.neon --memory-limit=-1", "stan-baseline": "@phpstan-baseline"}' composer.json
    container: badouralix/curl-jq

  - run: composer update nothing --no-interaction
    container: composer:latest

changesetTemplate:
  title: Add PHPStan baseline scripts to composer.json
  branch: add-phpstan-baseline-scripts
  commit:
    message: Add PHPStan baseline scripts to composer.json
  published: false
```
3. Run `src batch preview -f phpstan-baseline-composer.batch.yaml` as per guide

#### Expected behavior:

Command runs and gives me a URL to preview the batch changes

#### Actual behavior:

504 gateway time-out error, here is the command output:

![Screenshot 2021-11-28 at 5 17 21 pm](https://user-images.githubusercontent.com/253237/143778714-27101d8a-cc25-45c8-aa94-1348898ab0eb.png)

**Comment by eseliger** [comment]
Hello!
Is it correct that you are attempting to use Batch Changes with Sourcegraph Cloud at sourcegraph.com? 
In that case, BC is currently not supported on this platform. To give it a try, you need to run Sourcegraph enterprise currently. 
Sorry about the confusing error message though, I will get this fixed next week.

**Comment by courier-new** [comment]
Hey there, just following up on this!

What Erik said is true that we do not support Batch Changes on sourcegraph.com right now. However, the error you surfaced in this issue report actually appears to be due to the `repositoriesMatchingQuery` taking too long to resolve! If you adjust the query to `repositoriesMatchingQuery: file:composer.json count:100`, for example, we succeed with the resolving repositories step and later on fail with a much clearer error message:

```
{
     "message": "access to batch changes on Sourcegraph.com is currently not available",
     "path": [
       "createBatchSpec"
     ]
   }
```

We've filed a [separate issue](https://github.com/sourcegraph/sourcegraph/issues/30327) to track the 504 timeouts issue, since that's actually an API-wide problem, but I'm also going to make some modifications so that there's feedback even earlier that Batch Changes doesn't work on sourcegraph.com yet. 🙂
